docs/4.2/production.md

```diff
@@ -0,0 +1,269 @@
+| 3023      | Proxy      | SSH port clients connect to after authentication. A proxy will forward this connection to port `3022` on the destination node. | Allow inbound traffic from SSH clients. | Allow outbound traffic to SSH clients.
+| 3022      | Node       | SSH port to the Node Service. This is Teleport's equivalent of port `22` for SSH. | Allow inbound traffic from proxy host. | Allow outbound traffic to the proxy host.
+| 3025      | Auth       | SSH port used by the Auth Service to serve its Auth API to other nodes in a cluster. | Allow inbound connections from all cluster nodes. | Allow outbound traffic to cluster nodes.
+| 3024      | Proxy      | SSH port used to create "reverse SSH tunnels" from behind-firewall environments into a trusted proxy server. | <TODO> | <TODO>
+
+<!--TODO: Add several diagrams of firewall config examples-->
+
+## Installation
+
+We have a detailed [installation guide](/installation.md) which shows how to
+install all available binaries or [install from
+source](#../installation.md#installing-from-source). Reference that guide to learn
+the best way to install Teleport for your system and the come back here to
+finish your production install.
+
+### Filesystem Layout
+
+By default a Teleport node has the following files present. The location of all
+of them is configurable.
+
+| Default path                 | Purpose  |
+|------------------------------|----------|
+| `/etc/teleport.yaml` | Teleport configuration file.|
+| `/usr/local/bin/teleport` | Teleport daemon binary.|
+| `/usr/local/bin/tctl` | Teleport admin tool. It is only needed for auth servers.|
+| `/usr/local/bin/tsh` | Teleport CLI client tool. It is needed on any node that needs to connect to the cluster.|
+| `/var/lib/teleport` | Teleport data directory. Nodes keep their keys and certificates there. Auth servers store the audit log and the cluster keys there, but the audit log storage can be further configured via `auth_service` section in the config file.|
+
+## Running Teleport in Production
+
+### Systemd Unit File
+
+In production, we recommend starting teleport daemon via an init system like
+`systemd`. If systemd and unit files are new to you check out [this helpful guide](https://www.digitalocean.com/community/tutorials/understanding-systemd-units-and-unit-files). Here's the recommended Teleport service unit file for systemd.
+
+
+```yaml
+[Unit]
+Description=Teleport SSH Service
+After=network.target
+
+[Service]
+Type=simple
+Restart=on-failure
+# Set the nodes roles with the `--roles`
+# In most production environments you will not
+# want to run all three roles on a single host
+# --roles='proxy,auth,node' is the default value
+# if none is set
+ExecStart=/usr/local/bin/teleport start --roles=auth --config=/etc/teleport.yaml --pid-file=/var/run/teleport.pid
+ExecReload=/bin/kill -HUP $MAINPID
+PIDFile=/var/run/teleport.pid
+
+[Install]
+WantedBy=multi-user.target
+```
```

Comment:
I would probably link to the `systemd` file in the examples directory of the Teleport repo rather than copy/pasting the content here - if someone updates the example, they probably won't also update this copy and that's how things get out of sync.

Reply:
I've added three node, proxy and auth in https://github.com/gravitational/teleport/pull/3236/commits/482437539d5341edcef7ee3b0fd300fb9356a032 I know it's a little repetitive but I wanted to call out specific best practices per `systemd` unit.